src/event_model/basemodels/event_descriptor.py

```diff
@@ -0,0 +1,238 @@
+    ),
+]
+
+_ConstrainedDtypeNpStructure = List[Tuple[str, _ConstrainedDtype]]
+
+
+class DataKey(BaseModel):
+    """Describes the objects in the data property of Event documents"""
+
+    model_config = ConfigDict(extra="forbid")
+
+    limits: Annotated[
+        Union[Limits, None], Field(default=None, description="Epics limits.")
+    ]
+    choices: Annotated[
+        Union[List[str], None],
+        Field(default=None, description="Choices of enum value."),
+    ]
+    dims: Annotated[
+        Union[List[str], None],
+        Field(
+            default=None,
+            description="The names for dimensions of the data. Null or empty list "
+            "if scalar data",
+        ),
+    ]
+    dtype: Annotated[
+        Dtype,
+        Field(
+            description=(
+                "The type of the data in the event, given as a broad "
+                "JSON schema type."
+            )
+        ),
+    ]
+    dtype_numpy: Annotated[
+        Union[_ConstrainedDtype, _ConstrainedDtypeNpStructure, None],
+        Field(
+            default=None,
+            description=(
+                "The type of the data in the event, given as a "
+                "numpy dtype string (or, for structured dtypes, array)."
+            ),
+        ),
+    ]
+    external: Annotated[
+        Union[str, None],
+        Field(
+            default=None,
+            description="Where the data is stored if it is stored external "
+            "to the events",
+            pattern=r"^[A-Z]+:?",
+        ),
+    ]
+    object_name: Annotated[
+        Union[str, None],
+        Field(
+            default=None, description="The name of the object this key was pulled from."
+        ),
+    ]
+    precision: Annotated[
+        Union[int, None],
+        Field(
+            default=None,
+            description="Number of digits after decimal place if "
+            "a floating point number",
+        ),
+    ]
+    shape: Annotated[
+        List[int],
+        Field(description="The shape of the data.  Empty list indicates scalar data."),
+    ]
+    source: Annotated[
+        str, Field(description="The source (ex piece of hardware) of the data.")
+    ]
+    units: Annotated[
+        Union[str, None],
+        Field(default=None, description="Engineering units of the value"),
+    ]
+
+
+class PerObjectHint(BaseModel):
+    """The 'interesting' data keys for this device."""
+
+    model_config = ConfigDict(extra="forbid")
+
+    fields: Annotated[
+        Union[List[str], None],
+        Field(default=None, description="The 'interesting' data keys for this device."),
+    ]
+    NX_class: Annotated[
+        Union[str, None],
+        Field(
+            default=None,
+            description="The NeXus class definition for this device.",
+            pattern=r"^NX[A-Za-z_]+$",
+        ),
+    ]
+
+
+class Configuration(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+    data: Annotated[
+        Union[Dict[str, Any], None],
```

Comment:
...and now it's `Union[Dict, None]`.

Is there any way to persuade pydantic to treat `None` as `NotRequired` in the JSON schema?